policysync/processor_test.go

```diff
@@ -865,16 +870,23 @@ func getDialer(proto string) func(string, time.Duration) (net.Conn, error) {
 }
 
 const ListenerDir = "/tmp/felixut"
-const ListenerPath = "/tmp/felixut/policysync.sock"
-
-func openListener() net.Listener {
-	Expect(os.MkdirAll(ListenerDir, 0777)).To(Succeed())
-	_, err := os.Stat(ListenerPath)
-	if !os.IsNotExist(err) {
-		// file exists, try to delete it.
-		Expect(os.Remove(ListenerPath)).To(Succeed())
-	}
-	lis, err := net.Listen("unix", ListenerPath)
+const ListenerSocket = "policysync.sock"
+
+// Single source of random numbers insures repeated calls to makeTmpListenDir get unique paths.  But! source is not
+// threadsafe so we protect it with a Mutex. Not the most idiomatic, but simple and fast.
+var rng = rand.New(rand.NewSource(time.Now().UnixNano()))
+var rngLock = sync.Mutex{}
+
+func makeTmpListenerDir() string {
```

Comment:
I think there's a library function that does exactly what you're looking for here: `ioutil.TempDir()`